lib/ansible/modules/net_tools/netbox/netbox_interface.py

```diff
@@ -0,0 +1,317 @@
+      netbox_interface:
+        netbox_url: http://netbox.local
+        netbox_token: thisIsMyToken
+        data:
+          device: test100
+          name: GigabitEthernet1
+        state: present
+    - name: Delete interface within netbox
+      netbox_interface:
+        netbox_url: http://netbox.local
+        netbox_token: thisIsMyToken
+        data:
+          device: test100
+          name: GigabitEthernet1
+        state: absent
+    - name: Create LAG with several specified options
+      netbox_interface:
+        netbox_url: http://netbox.local
+        netbox_token: thisIsMyToken
+        data:
+          device: test100
+          name: port-channel1
+          form_factor: Link Aggregation Group (LAG)
+          mtu: 1600
+          mgmt_only: false
+          mode: Access
+        state: present
+    - name: Create interface and assign it to parent LAG
+      netbox_interface:
+        netbox_url: http://netbox.local
+        netbox_token: thisIsMyToken
+        data:
+          device: test100
+          name: GigabitEthernet1
+          enabled: false
+          form_factor: 1000Base-t (1GE)
+          lag:
+            name: port-channel1
+          mtu: 1600
+          mgmt_only: false
+          mode: Access
+        state: present
+    - name: Create interface as a trunk port
+      netbox_interface:
+        netbox_url: http://netbox.local
+        netbox_token: thisIsMyToken
+        data:
+          device: test100
+          name: GigabitEthernet25
+          enabled: false
+          form_factor: 1000Base-t (1GE)
+          untagged_vlan:
+            name: Wireless
+            site: Test Site
+          tagged_vlans:
+            - name: Data
+              site: Test Site
+            - name: VoIP
+              site: Test Site
+          mtu: 1600
+          mgmt_only: true
+          mode: Tagged
+        state: present
+"""
+
+RETURN = r"""
+interface:
+  description: Serialized object as created or already existent within Netbox
+  returned: on creation
+  type: dict
+msg:
+  description: Message indicating failure or info about what has been achieved
+  returned: always
+  type: str
+"""
+
+import json
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.net_tools.netbox.netbox_utils import find_ids, normalize_data, INTF_FORM_FACTOR, INTF_MODE
+from ansible.module_utils.compat import ipaddress
+from ansible.module_utils._text import to_text
+
+
+PYNETBOX_IMP_ERR = None
+try:
+    import pynetbox
+    HAS_PYNETBOX = True
+except ImportError:
+    PYNETBOX_IMP_ERR = traceback.format_exc()
+    HAS_PYNETBOX = False
+
+
+def main():
+    """
+    Main entry point for module execution
+    """
+    argument_spec = dict(
+        netbox_url=dict(type="str", required=True),
+        netbox_token=dict(type="str", required=True, no_log=True),
+        data=dict(type="dict", required=True),
```

Comment:
Should be able to define the structure of `data` here, something like https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/f5/bigiq_application_fastl4_tcp.py#L646-L653

Reply:
Fixed this as well.